source/shared/cpp/ObjectModel/ParseContext.h

```diff
@@ -21,16 +21,24 @@ namespace AdaptiveSharedNamespace
         std::shared_ptr<ElementParserRegistration> elementParserRegistration;
         std::shared_ptr<ActionParserRegistration> actionParserRegistration;
         std::vector<std::shared_ptr<AdaptiveCardParseWarning>> warnings;
-
         // Push/PopElement are used during parsing to track the tree structure of a card.
         void PushElement(const std::string& idJsonProperty,
                          const AdaptiveSharedNamespace::InternalId& internalId,
                          const bool isFallback = false);
         void PopElement();
 
+        void SetLanguage(const std::string& value);
+        std::string GetLanguage() const;
+
+        ContainerStyle GetParentalContainerStyle() const;
+        AdaptiveSharedNamespace::InternalId GetIDOfParentWithPadding(void)const;
```

Comment:
>const [](start = 74, length = 5)

please use `clang-format` :)

Reply:
i'm using it. :-)
i guess i might have edited this on mac.

---
In reply to: [259079305](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259079305) [](ancestors = 259079305)

Reply:
i edit using vim; then use vs to compile. i've thought when vs update its buffer, it will trigger clang-format. it wasn't the case. i just installed clang-format plugin for vim. Thanks!

---
In reply to: [259096326](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259096326) [](ancestors = 259096326,259079305)

Reply:
the closest you can get in VS is the *Format document on Save* [plugin](https://marketplace.visualstudio.com/items?itemName=mynkow.FormatdocumentonSave)

---
In reply to: [259198433](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259198433) [](ancestors = 259198433,259096326,259079305)

Reply:
awesome

---
In reply to: [259540302](https://github.com/Microsoft/AdaptiveCards/pull/2406#discussion_r259540302) [](ancestors = 259540302,259198433,259096326,259079305)